apps/mobile/src/features/ConfirmTx/hooks/useTransactionData.ts

```diff
@@ -0,0 +1,16 @@
+import { useTransactionsGetTransactionByIdV1Query } from '@safe-global/store/gateway/AUTO_GENERATED/transactions'
+import { useDefinedActiveSafe } from '@/src/store/hooks/activeSafe'
+
+export const useTransactionData = (txId: string) => {
```

Comment:
nit: this fetches txDetails afaik. There is also txData which could be confusing so it might be better to rename this to `useTransactionDetails`

Reply:
no, it calls this endpoint: https://safe-client.staging.5afe.dev/v1/chains/11155111/transactions/0xb0133306ef864ae3d294c9dc9720c03243eada7aa4e8506e7fded16ad0e807ae which contains everything about the tx including txDetails